File: app/src/main/java/com/martinlaizg/geofind/views/fragment/play/PlayThermFragment.java
package com.martinlaizg.geofind.views.fragment.play;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.cardview.widget.CardView;

import com.martinlaizg.geofind.R;

import butterknife.BindView;
import butterknife.ButterKnife;

public class PlayThermFragment
		extends PlayTourFragment {

	private static final String TAG = PlayThermFragment.class.getSimpleName();

	@BindView(R.id.therm_card)
	CardView therm_card;

	private float lastDistance = 0;

	@Nullable
	@Override
	public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
			@Nullable Bundle savedInstanceState) {
		View view = inflater.inflate(R.layout.fragment_play_therm, container, false);
		ButterKnife.bind(this, view);
		return view;
	}

	@Override
	protected String TAG() {
		return TAG;
	}

	@Override
	void updateView() {
		if(distance > lastDistance) {
			therm_card.setCardBackgroundColor(getResources().getColor(R.color.red, null));
		} else {
			therm_card.setCardBackgroundColor(getResources().getColor(R.color.green, null));
		}
		lastDistance = distance;
	}
}

File: app/src/main/java/com/martinlaizg/geofind/views/fragment/play/PlayThermFragment.kt
package com.martinlaizg.geofind.views.fragment.play

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.cardview.widget.CardView
import com.martinlaizg.geofind.R

class PlayThermFragment : PlayTourFragment() {

	private var thermCard: CardView? = null
	private var lastDistance = 0f

	override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
	                          savedInstanceState: Bundle?): View? {
		val view = inflater.inflate(R.layout.fragment_play_therm, container, false)
		thermCard = view.findViewById(R.id.therm_card)
		return view
	}

	override fun updateView() {
		if (distance > lastDistance) {
			thermCard!!.setCardBackgroundColor(
					resources.getColor(android.R.color.holo_red_dark, null))
		} else {
			thermCard!!.setCardBackgroundColor(
					resources.getColor(android.R.color.holo_green_dark, null))
		}
		lastDistance = distance
	}
}